### PR TITLE
multi: update to lnd with MuSig2

### PR DESCRIPTION
**Pull request by bhandras**
Takeover of: https://github.com/lightninglabs/lndclient/pull/97

### CODE REVIEW

**Comment by bhandras** [comment]
Updated to latest musig2 API.

**Comment by bhandras** [comment]
One addition over the original PR is that I added `MuSig2TaprootTweakOpt` too.

File: signer_client.go [hunk 1]
```diff
@@ -344,3 +373,166 @@ func (s *signerClient) DeriveSharedKey(ctx context.Context,
...
+func MuSigNonceOpt(nonces [][musig2.PubNonceSize]byte) MuSig2SessionOpts {
+	return func(s *signrpc.MuSig2SessionRequest) {
+		s.OtherSignerPublicNonces = noncesToBytes(nonces)
+	}
+}
+
+// MuSig2TaprootTweakOpt adds an optional taproot tweak to the musig session
+// request.
+func MuSig2TaprootTweakOpt(
+	scriptRoot []byte, keySpendOnly bool) MuSig2SessionOpts {
```
Comment (guggero) [inline signer_client.go:402]: nit: `scriptRoot` param could fit on previous line.

File: signer_client.go [hunk 2]
```diff
@@ -344,3 +373,166 @@ func (s *signerClient) DeriveSharedKey(ctx context.Context,
...
+
+	for i := range nonces {
+		nonceBytes[i] = nonces[i][:]
+	}
+
+	return nonceBytes
+}
+
+// MuSig2NonceOpt adds an optional set of nonces to a musig session request.
+func MuSigNonceOpt(nonces [][musig2.PubNonceSize]byte) MuSig2SessionOpts {
```
Comment (guggero) [inline signer_client.go:393]: nit: `2` missing in function name.

File: signer_client.go [hunk 3]
```diff
@@ -344,3 +373,166 @@ func (s *signerClient) DeriveSharedKey(ctx context.Context,
...
+// cleanup bool should be set.
+func (s *signerClient) MuSig2Sign(ctx context.Context, sessionID [32]byte,
+	message [32]byte, cleanup bool) ([]byte, error) {
+
+	req := &signrpc.MuSig2SignRequest{
+		SessionId:     sessionID[:],
+		MessageDigest: message[:],
+		Cleanup:       cleanup,
+	}
+	rpcCtx, cancel := context.WithTimeout(ctx, s.timeout)
```
Comment (guggero) [inline signer_client.go:505]: nit: add newline before to match the other methods?

**Review by guggero — APPROVED** [review]
Nice, LGTM :tada: 

Only a few nits.

File: signer_client.go (hunk 1 above)
Comment (bhandras) [reply signer_client.go]: done

File: signer_client.go (hunk 3 above)
Comment (bhandras) [reply signer_client.go:506]: done

**Pull request merged**